pkg/controllers/networkpolicy/controller.go

```diff
@@ -764,3 +782,95 @@ func (r *NetworkPolicyReconciler) resetWatchers() error {
+// controller indexers
+func (r *NetworkPolicyReconciler) removeIndexerObjectsByAccount(namespacedName string) error {
+	r.Log.Info("Clearing indexers for the account", "account", namespacedName)
+	atSgs, err := r.appliedToSGIndexer.ByIndex(addrAppliedToIndexerByAccountId, namespacedName)
+	if err != nil {
+		r.Log.Error(err, "failed to get appliedToGroups from indexer", "account", namespacedName)
+	}
+
+	if len(atSgs) > 0 {
+		for _, obj := range atSgs {
+			atSg, ok := obj.(*appliedToSecurityGroup)
+			if !ok {
+				continue
+			}
+
+			// Delete NPs based on AT.
+			nps, err := r.networkPolicyIndexer.ByIndex(networkPolicyIndexerByAppliedToGrp, atSg.id.Name)
+			if err != nil {
+				r.Log.Error(err, "failed to get np from indexer", "appliedToGroup", atSg.id.Name)
+			}
+			for _, obj := range nps {
+				np, ok := obj.(*networkPolicy)
+				if !ok {
+					continue
+				}
+
+				r.Log.V(1).Info("Deleting np from indexer", "np", np.Name)
+				if err := r.networkPolicyIndexer.Delete(np); err != nil {
+					r.Log.Error(err, "failed to delete np from indexer", "np", np.Name)
+				}
+			}
+
+			// Delete CloudRules based on AT.
+			rules, err := r.cloudRuleIndexer.ByIndex(cloudRuleIndexerByAppliedToGrp, atSg.id.CloudResourceID.String())
+			if err != nil {
+				r.Log.Error(err, "failed to get cloud rules from indexer", "appliedToGroup", atSg.id.CloudResourceID.String())
+			}
+			for _, obj := range rules {
+				rule, ok := obj.(*cloudresource.CloudRule)
+				if !ok {
+					continue
+				}
+
+				r.Log.V(1).Info("Deleting cloud rule from indexer", "rule", rule.NpNamespacedName)
+				if err := r.cloudRuleIndexer.Delete(rule); err != nil {
+					r.Log.Error(err, "failed to delete cloud rule from indexer",
+						"appliedToGroup", atSg.id.CloudResourceID.String(), "rule", rule.NpNamespacedName)
+				}
+			}
+
+			// Remove AT group from retryQueue and pendingDeleteGroups.
+			uName := getGroupUniqueName(atSg.id.CloudResourceID.String(), false)
+			uGroupName := getGroupUniqueName(atSg.id.Name, false)
+			r.retryQueue.Remove(uName)
+			r.pendingDeleteGroups.Remove(uGroupName)
+
+			// Delete AT.
+			r.Log.V(1).Info("Deleting appliedToGroup from indexer", "appliedToGroup", atSg.id.Name)
+			if err := r.appliedToSGIndexer.Delete(atSg); err != nil {
+				r.Log.Error(err, "failed to delete appliedToGroup from indexer", "atSg", atSg.id.Name)
+			}
+		}
+	}
+
+	// Delete AG.
+	agSgs, err := r.addrSGIndexer.ByIndex(addrAppliedToIndexerByAccountId, namespacedName)
+	if err != nil {
+		r.Log.Error(err, "failed to get addressGroups from indexer", "account", namespacedName)
+	}
+	if len(agSgs) > 0 {
+		for _, obj := range agSgs {
+			agSg, ok := obj.(*addrSecurityGroup)
+			if !ok {
+				continue
+			}
+
+			// Remove AG group from retryQueue and pendingDeleteGroups.
+			uName := getGroupUniqueName(agSg.id.CloudResourceID.String(), true)
+			uGroupName := getGroupUniqueName(agSg.id.Name, true)
+			r.retryQueue.Remove(uName)
```

Comment:
This wont work. In case of delete, we remove the item from indexer first, and then put in retry queue. So, you wont find the item for lookup in retry queue. You might have to iterate over items in the queue and delete all matching account-id.